core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Builds a new {@link ServerListener}.
+ * */
+public class ServerListenerBuilder {
+
+    /**
+     * {@link Runnable}s invoked on when the {@link Server} is starting.
+     * */
+    private List<Runnable> onStartings = new ArrayList<>();
```

Comment:
How about watching the names of server listener?

`serverStartingCallbacks` etc